python/ray/serve/controller.py

```diff
@@ -358,6 +358,22 @@ def deploy(
 
         return goal_id, updating
 
+    def deploy_group(
+        self, deployment_args_list: List[Dict]
+    ) -> List[Tuple[Optional[GoalId], bool]]:
+        """
+        Takes in a list of dictionaries that contain keyword arguments for the
+        controller's deploy() function. Calls deploy on all the argument
+        dictionaries in the list. Effectively executes an atomic deploy on a
+        group of deployments.
+        """
+
+        update_goals: List[Tuple[Optional[GoalId], bool]] = []
+        for deployment_args in deployment_args_list:
```

Comment:
it's worth considering as a follow-up to commit all deployment infos as a batch to controller then let it handle replicas for all in each update() cycle, rather than blocking and waiting for each at a time to reduce e2e latency.

Reply:
With my PR we can just poll and watch all of the statuses instead.

Reply:
that's much better. I felt our current controller logic and data structured not suited for optimizing for group deployments at all :/ looking at your pr ..

Reply:
yup